examples/seismic/self_adjoint/time_blocking.ipynb

```diff
@@ -0,0 +1,1365 @@
+{
```

Comment:
<p><code>update2</code>  is syntactically identical to <code>update1</code>  . Same story for <code>v1_term</code>  and <code>v2_term</code>   as well as <code>stencil1</code>  and <code>stencil2</code>  ....</p><p> </p><p>I think you only need to construct <em>one</em> <code>Eq</code>  , and to get the other one you just do <code>stencil.subs({u1: u2, v1: v2})</code>  or something along these lines.</p><p> </p><p><strong>If</strong> the two Operators produce the same exact code, you actually don't even need that. You may reuse the same Operator, the second time called as <code>op.apply(u1=u2, v1=v2)</code>   . I just don't remember whether <code>save=Buffer(M)</code> makes the compiler generate the same exact code as <code>save=nt</code>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/devitocodes/devito/pull/1424/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Reply:
Different `time` tho. I htink you need to use a stencil with the different `time` or doesn't work.

Reply:
Ah, you mean before building the operator ... I will test

Reply:
As notes on slack, this may complicate the exposition, even as it simplifies the code. I dont think it is the right way to go, but can live with your call.

Reply:
I don't think a `make_stencil` function (instead of subs) would complicate exposition (in fact, the opposite? it's the same stencil expression after all), but I can live with the way it is now. Perhaps you may add a comment that you can avoid some duplicate code either via subs or with a function?

Reply:
New comment before the first operator:

## Note on code duplication

The stencils for the two operators you see below is exactly the same, the only significant difference is that we use two different ```TimeFunction```s. We could therefore reduce code duplication in two ways:

1. Use the placeholder design pattern and ```stencil.subs``` to substitude the appropriate ```TimeFunction```. 
Please see the FAQ for more information [https://github.com/devitocodes/devito/wiki/FAQ#how-are-abstractions-used-in-the-seismic-examples](https://github.com/devitocodes/devito/wiki/FAQ#how-are-abstractions-used-in-the-seismic-examples)

2. Write a function and use it to build the stencils

To increase the clarity of the exposition below, We do neither of these and duplicate the stencil code.